### PR TITLE
More field names and a method

### PR DESCRIPTION
**Pull request by oscargus**
More field names in `FieldName` and a method `FieldName.orFields(FieldName.YEAR, FieldName.DATE)` to generate `"year/date"`.

- [x] Tests created for changes

### CODE REVIEW

File: src/main/java/net/sf/jabref/model/entry/FieldName.java [hunk 1]
```diff
@@ -42,16 +43,35 @@
...
     public static final String URLDATE = "urldate";
+    public static final String VERSION = "version";
     public static final String VOLUME = "volume";
     public static final String YEAR = "year";
+    public static final String YEARFILED = "yearfiled";
+
+    // JabRef internal field names
+    public static final String OWNER = "owner";
+    public static final String TIMESTAMP = "timestamp"; // Not the actual field name, but the default value
+    public static final String NUMBER_COL = "#";
```
Comment (matthiasgeiger) [inline src/main/java/net/sf/jabref/model/entry/FieldName.java]: Is this a real "field" with a field name and used as such?

Comment (matthiasgeiger) [inline src/main/java/net/sf/jabref/model/entry/FieldName.java]: Okay it is... https://github.com/JabRef/jabref/pull/1627/files#diff-59eb0b0c47b8c7b57710bde78ec9f349L215

**Comment by matthiasgeiger** [comment]
LGTM!

File: src/main/java/net/sf/jabref/model/entry/FieldName.java (hunk 1 above)
Comment (oscargus) [inline src/main/java/net/sf/jabref/model/entry/FieldName.java]: I think so, yes.

From InternalBibtexFields:

``` java 
         dummy = new BibtexSingleField(FieldName.NUMBER_COL, false, 32);
         dummy.setPrivate();
         dummy.setWriteable(false);
         dummy.setDisplayable(false);
         add(dummy);
```

**Pull request merged**